src/main/java/com/bnorm/auto/dispatch/internal/Writer.java

```diff
@@ -108,9 +112,38 @@ public static JavaFile write(MethodClassDescriptor methodClassDescriptor) {
         codeBlockBuilder.addStatement("throw new NullPointerException($S)", "dispatch value == null");
         first = false;
       }
-      for (MultiDescriptor descriptor : methodDescriptor.multiMethods()) {
+
+      // Work with multiMethod dispatch values as a list
+      // This allows a multiMethod to accept several dispatch values
+      Function<MultiDescriptor, List<?>> toDispatchValues = descriptor ->
+          descriptor.value().getValue() instanceof List<?> ? (List<?>) descriptor.value().getValue()
+                                                           : Collections.singletonList(descriptor.value().getValue());
+
+      // Dispatch to multiMethods from the most specific to the most accepting
+      List<MultiDescriptor> multiMethods = new ArrayList<>(methodDescriptor.multiMethods());
+      multiMethods.sort((m1, m2) -> {
+        List<?> values1 = toDispatchValues.apply(m1);
+        List<?> values2 = toDispatchValues.apply(m2);
+
+        return Integer.compare(values1.size(), values2.size());
+      });
+
+      for (MultiDescriptor descriptor : multiMethods) {
         String prefix = first ? "else " : "";
-        codeBlockBuilder.nextControlFlow(prefix + "if (value == $L)", descriptor.value());
+
+        // will always have at least one value
+        List<?> dispatchValues = toDispatchValues.apply(descriptor);
+        if (dispatchValues.size() == 1) {
+          String value =
+              descriptor.value().getValue() instanceof List<?> ? ((List<?>) descriptor.value().getValue()).get(0)
+                                                                                                          .toString()
+                                                               : descriptor.value().toString();
+          codeBlockBuilder.nextControlFlow(prefix + "if (value == $L)", value);
+        } else {
+          String valuesString = dispatchValues.stream().map(Object::toString).reduce((v1, v2) -> v1 + ", " + v2).get();
+          codeBlockBuilder.nextControlFlow(prefix + "if ($T.asList($L).contains(value))", Arrays.class, valuesString);
```

Comment:
This is going to create a new List every time it's called.  I would prefer to see a bunch of ORs instead.  Otherwise the performance and memory/gc hit is going to be massive.

The other option would be to create a static final field for this.  That would be another acceptable option.